controllers/healthcheck_controller.go

```diff
@@ -347,6 +503,59 @@ func (r *HealthCheckReconciler) watchWorkflowReschedule(ctx context.Context, log
 	return nil
 }
 
+func (r *HealthCheckReconciler) watchRemedyWorkflow(log logr.Logger, wfNamespace, wfName string, hc *activemonitorv1alpha1.HealthCheck) error {
+	var now metav1.Time
+	then := metav1.Time{Time: time.Now()}
+	//repeatAfterSec := hc.Spec.RepeatAfterSec
+	for {
```

Comment:
Can we use https://github.com/keikoproj/inverse-exp-backoff instead of a simple loop here?

Reply:
sure..

Reply:
I checked the inverse-exp-backoff.. the challange in using it here would be that although we retry in a for loop here there is an activeDeadlineSeconds that will be set in the workflow based on the RepeatAfterSec parameter. So after reaching the activeDeadlineSeconds the pods in the workflow will be cleaned up. So we need to update https://github.com/keikoproj/inverse-exp-backoff to work based on a timeout and then use it. Let me know if that is okay.

Reply:
The reason we need to implement a timeout based inverse-exp-backoff is each workflow might have a different RepeatAfterSec interval that would be configured so we cannot have a generic value on the maxretries and other parameters, initialdelay and factor in inverse-exp-backoff.

Reply:
Interesting. Can you file a feature request in inverse-exp-backoff? Instead of failing based on total retries, it should also support (maybe a different API) for failing based on on total time.

Reply:
sure

Reply:
https://github.com/keikoproj/inverse-exp-backoff/issues/3